### ISSUE TITLE
Wrappers not working as expected with PISCES 1D parameter sensitiviy analysis workflow, AS 3.14.0

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#941]
In GitLab by @mgali on Feb 6, 2023, 12:36

This is the follow-up of another issue:

[Modified workflow to construct a surrogate model for sensitivity analyses](https://earth.bsc.es/gitlab/cp/genetic_algorithm_pisces1d/-/issues/10)

It may also be related to 

[Horizontal-vertical wrapper fine tuning, autosubmit issue #640](https://earth.bsc.es/gitlab/es/autosubmit/-/issues/640)

I copy the previous message below:

Hi @dbeltrankyl and @mcastril ,

Update:

- I launched **a45b** (account bsc32882) following your specifications, debug queue. 
- I launched **a5mr** (account bsc32340) without vertical wrappers (as I used to do), bsc_es queue.

In both cases I see some chunks were completed, but most failed. Some before completing a single SIM, others after completing a few.

Failures seem related to hitting the wallclock, although I cannot always see the TIME LIMIT message in the logs. I also tried to change the jobs wallclock from 10 to 20 minutes (**a44w** on bsc32882). Same issues. In summary: it seems to me than when jobs in a chunk start running before hitting the wallclock, some of them make it to the target (not all). If the first sim hits the wallclock, then the whole chunk is failed.

I noted some different behaviours compared to previous runs with AS 3.13.0 (Nov 2021): this time I see jobs marked as running but they are actually not! I know because in the runtime folders nothing is happening that indicates a running sim. Just a few jobs are simultaneously queueing, where I would expect all 100 jobs (members) to run or queue simultaneously.

We can assume the nemo simulation environment and executable themselves are fine: I changed nothing compared to older experiments, and I just managed to complete another experiment (**a5mj**) with 2000 sequential sims without failures.

I'll copy this to autosubmit.

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 6, 2023, 12:46

mentioned in issue cp/genetic_algorithm_pisces1d#10

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 6, 2023, 12:46

Hello, 

I think there is something wrong with the template itself. When I look at the logs of each chunk ( from the autosubmit GUI, you can click to each job and see it's log ) I see this:


/esarchive/autosubmit/a5mr/tmp/LOG_a5mr/a5mr_19600101_fc0044_1_SIM.20230203174754.out 

The error is the same for each member I revised


```
ew = pd.DataFrame(data = names.iloc[[row]].values, columns = names.keys(), index = [row])
new.to_csv(str(row)+'_params.csv')
END
++ python - /gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001_3500.csv /gpfs/scratch/bsc32/bsc32340/a5mr/EXP00/0042/namelist_pisces_cfg 0042
Traceback (most recent call last):
  File "<stdin>", line 10, in <module>
  File "/apps/PYTHON/3.6.1/INTEL/lib/python3.6/site-packages/pandas/io/parsers.py", line 676, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "/apps/PYTHON/3.6.1/INTEL/lib/python3.6/site-packages/pandas/io/parsers.py", line 448, in _read
    parser = TextFileReader(fp_or_buf, **kwds)
  File "/apps/PYTHON/3.6.1/INTEL/lib/python3.6/site-packages/pandas/io/parsers.py", line 880, in __init__
    self._make_engine(self.engine)
  File "/apps/PYTHON/3.6.1/INTEL/lib/python3.6/site-packages/pandas/io/parsers.py", line 1114, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/apps/PYTHON/3.6.1/INTEL/lib/python3.6/site-packages/pandas/io/parsers.py", line 1891, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 374, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 674, in pandas._libs.parsers.TextReader._setup_parser_source
FileNotFoundError: [Errno 2] File /gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001_3500.csv does not exist: '/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001_3500.csv'
+ nls=
```

Edit: I manually revised it, and indeed this doesn't exists:

```
bsc32070@login2:~> cat /gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001_3500.csv
cat: /gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001_3500.csv: No such file or directory
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 6, 2023, 12:53

I think there is a typo in the template since:

This doesn't exist 

`cat: /gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001_3500.csv: No such file or directory`

This exists: 

`bsc32070@login2:~> ls /gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001-3500.csv
/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001-3500.csv
`

( "_" char instead of "-" char)

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 6, 2023, 13:20

Hi @dbeltrankyl ,

Where do you see the file with "underscore"_" instead of "-"?

I grep-ed all my templates and couldn't see it:

```
mgalitap@ bscearth302:/esarchive/autosubmit> grep sobol a5mr/proj/autosubmit/templates/*.sh
a5mr/proj/autosubmit/templates/pre.sh:PARAMETERS=/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001-3500.csv
a5mr/proj/autosubmit/templates/sim.sh:PARAMETERS=/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001-3500.csv
mgalitap@ bscearth302:/esarchive/autosubmit> grep sobol a45b/proj/autosubmit/templates/*.sh
a45b/proj/autosubmit/templates/pre.sh:PARAMETERS=/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims8501-10000.csv
a45b/proj/autosubmit/templates/sim.sh:PARAMETERS=/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims8501-10000.csv
mgalitap@ bscearth302:/esarchive/autosubmit> grep sobol a44w/proj/autosubmit/templates/*.sh
a44w/proj/autosubmit/templates/pre.sh:PARAMETERS=/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001-3500.csv
a44w/proj/autosubmit/templates/sim.sh:PARAMETERS=/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001-3500.csv
mgalitap@ bscearth302:/esarchive/autosubmit> grep sobol a5mj/proj/autosubmit/templates/*.sh
a5mj/proj/autosubmit/templates/pre.sh:PARAMETERS=/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001-3500.csv
a5mj/proj/autosubmit/templates/sim.sh:PARAMETERS=/gpfs/scratch/bsc32/bsc32340/pisces1d/parameters/sobolSeq_N_10000_P_9_2021-11-24_sims2001-3500.csv

```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 6, 2023, 13:50

Hello, 

Sorry, seems that the gui Didn't have the last log, 

The last log had this issue:

```
(Requested nodes are busy)
+ mpirun -np 1 ./nemo
srun: Job 27289028 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27289028 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27289028 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27289028 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27289028 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27289028 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27289028 step creation still disabled, retrying (Requested nodes are busy)
srun: error: Unable to create step for job 27289028: Job/step already completing or completed
```

I think it may be related to the node memory, but I have no experience with this. I think  @mcastril may be able to help you better than I.

Perhaps you can try to limit the horizontal (not sure if the vertical, too ) size using the  {min/max }wrapped_{h/v} to solve this issue, but not sure about the limits.

Also, if you tune the PROCESSORS_PER_NODE=X in platforms.conf you can allocate more or less nodes. Usingm ore nodes can also help

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 6, 2023, 13:59

Okay, forgot about the memory I think this maybe the real issue.


The wrapper is allocating  89 nodes using bsc_es queue but only 50 nodes ( 16 in debug) are allowed in bsc_es

| BSC | 50 (2400) | 48 h |
|-----|-----------|------|

If you limit the horizontal size to 50 it should work. MAX_WRAPPED_H = 50.

What is happening is that some jobs are waiting until the nodes are released ( by other jobs of your experiment) and this causes a mess in the wallclock of some jobs as they are counting already as running but they did not start yet due to the nodes being busy.

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 6, 2023, 14:17

Thanks a lot @dbeltrankyl ,

This explanation makes sense, intuitively, although I need to read better the docs to understand how processors and nodes are organized in h and v wrappers.

In the meantime I'll try what you suggest. In fact my workflow would need only 3 nodes (48 + 48 + 4 = 100 parallel jobs). It's perhaps not casual that 96 sims (hence 96 members) failed and only 4 could proceed in exps a44w and a45b.

So the key may be asking for at least 34 processors per node with 3 nodes. Maybe even better to ask for the whole node (48).

I don't know how to tune the number of v wrappers here. You suggest 2-12 for MAX_WRAPPED_V, but in this workflow my chunks (vertical) are 15 to 20... I can also try again without v wrappers.

Your view @mcastril ?

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 6, 2023, 14:55

Additional note:

In the previous exps, PROCESSORS_PER_NODE=48. But it seems that less than 48 processors were being allocated, otherwise there wouldn't have been the problem with jobs waiting, right? 

I re-launched the experiments with MAX_WRAPPED_H = 3 to try to ensure we are using as many processors as possible in a given node. Not sure what I did is sensible

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 6, 2023, 15:23

Hello @mgali ,

I've doing some tests, in a dummy copy 

> In the previous exps, PROCESSORS_PER_NODE=48. But it seems that less than 48 processors were being allocated, otherwise there wouldn't have been the problem with jobs waiting, right?

You're right. The issue is that you don't have a TASKS = 1 specified in the jobs.conf. This was causing to autosubmit to request 50 nodes

```
[SIM]
TASKS = 1
```

Autosubmit wrapper thinks SIM requires 48 cores instead of 1 if not specified. 

Then in the autosubmit.conf, you now have only a horizontal defined, you can define a horizontal-vertical:

```
[wrapper]
TYPE = horizontal-vertical
JOBS_IN_WRAPPER = SIM
#MACHINEFILES = STANDARD remove this
MAX_WRAPPED_H = 720 or 2400 ( debug vs bsc_es)
```

then in platforms.conf

if you want to use debug:

```
[marenostrum4]
# Queue type. Options: ps, SGE, LSF, SLURM, PBS, eceaccess
TYPE = slurm
HOST = mn1.bsc.es
PROJECT = bsc32
USER = bsc32### <- your user
SCRATCH_DIR = /gpfs/scratch
ADD_PROJECT_TO_HOST = false
QUEUE = debug
PROCESSORS_PER_NODE = 48
MAX_PROCESSORS = 720
MAX_WALLCLOCK = 02:00
```

if you want to use bsc_es

```
[marenostrum4]
# Queue type. Options: ps, SGE, LSF, SLURM, PBS, eceaccess
TYPE = slurm
HOST = mn1.bsc.es
PROJECT = bsc32
USER = bsc32### <- your user
SCRATCH_DIR = /gpfs/scratch
ADD_PROJECT_TO_HOST = false
QUEUE = bsc_es
PROCESSORS_PER_NODE = 48
MAX_PROCESSORS = 2400
MAX_WALLCLOCK = 48:00
```

If you think there can be memory issues, you can reduce `
PROCESSORS_PER_NODE = 24` 

and decrease the max_wrapped_h value


```
[wrapper]
TYPE = horizontal-vertical
JOBS_IN_WRAPPER = SIM
#MACHINEFILES = STANDARD remove this
MAX_WRAPPED_H = 360 or 1200 ( debug vs bsc_es when processor_per_node = 24 )
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 6, 2023, 15:26

Now my squeue has:

```
bsc32070@login2:~> squeue
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
          27316294      main a057_AST bsc32070 CG       0:17      2 s12r2b[52,68]
```

and sacct:

```
JobID           JobName  Partition    Account  AllocCPUS      State ExitCode 
------------ ---------- ---------- ---------- ---------- ---------- -------- 
27316288     a057_ASTh+       main      bsc32         96  COMPLETED      0:0 
27316288.ba+      batch                 bsc32         48  COMPLETED      0:0 
27316288.ex+     extern                 bsc32         96  COMPLETED      0:0 

```

So the values seems correct

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 7, 2023, 09:17

Great! Launching a new exp now with all your suggestions. Thanks a lot!Great! Launching a new exp now with all your suggestions. Thanks a lot!

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 7, 2023, 10:54

Hola @dbeltrankyl ,

Experiment ***a458*** shows 97 failed sims, 3 completed, although at the beginning it displayed 100 sim as running.

Seems further adjustments in wrappers parameters are needed. Should I try with horizontal only? thanks

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 7, 2023, 17:11

Update: I relaunched **a458** with these changes in 

platforms.conf

```
[marenostrum4]
PROCESSORS_PER_NODE = 24
```

autosubmit.conf
```
[wrapper]
MAX_WRAPPED_H = 360
```

The experiment got stuck after INI. All 100 first jobs ready but not even queueing. I see this in the LOGS, *.err

```
Traceback (most recent call last):
  File "/scratch/slurm/job27332798/slurm_script", line 120, in <module>
    all_cores.append(all_nodes[idx])
IndexError: list index out of range
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 8, 2023, 09:40

Hello, 

I'm sorry, that should've worked, but there seems to be a bug. I'll solve it for 3.15.0b. I guess it wasn't detected because nobody tried to use it like that. 

We can achieve the same by reducing the horizontal size to 25, so only half node is used. It means that 4 wrappers will be made if you have 100 members, but since it is a horizontal-vertical, it shouldn't be an 
issue. 

So the config would be:

Platforms.conf

change from 24 to `PROCESSORS_PER_NODE = 48`


```
Wrapper

change MAX_WRAPPED_H = 25
```


I tested this approach with a dummy, and the jobs finished correctly, but they are dummy jobs that don't use memory.

Edit: When you start the experiment, let me know, and I'll monitor it throughout the day

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 8, 2023, 13:40

Thanks for checking! And happy to learn these tests may help you add a new feature to AS 3.15.0. Will test your settings in a real exp right now and tell you if they worked. Cheers,

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 8, 2023, 13:55

Telling you when running

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 8, 2023, 14:53

Well, I re-launched a458 but did a stupid thing, and emptied the tmp directory thinking it just had old stuff. So AS cannot retrieve logs now.

I "cloned" a458 into new exp **a5nd**. This is the one you can safely look at. Fingers crossed

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 8, 2023, 15:07

Hello, 

Yes, autosubmit needs to have tmp/ASLOGS and  tmp/LOG_<expid>, if you recreate the folders it should work again, 

I'm looking directly at the remote logs, and now it looks much better

```
bsc32070@login2:/gpfs/scratch/bsc32/bsc32340/a458/LOG_a458> tail -n 5 a458_19600101_fc00*_1_SIM.cmd.err.*
==> a458_19600101_fc0000_1_SIM.cmd.err.0 <==
+ echo 1675862885
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0000_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0001_1_SIM.cmd.err.0 <==
+ echo 1675861882
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0001_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0002_1_SIM.cmd.err.0 <==
+ echo 1675861382
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0002_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0003_1_SIM.cmd.err.0 <==
+ echo 1675861140
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0003_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0004_1_SIM.cmd.err.0 <==
+ echo 1675861714
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0004_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0005_1_SIM.cmd.err.0 <==
+ echo 1675864388
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0005_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0006_1_SIM.cmd.err.0 <==
+ echo 1675864889
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0006_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0007_1_SIM.cmd.err.0 <==
+ echo 1675861548
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0007_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0008_1_SIM.cmd.err.0 <==
+ echo 1675863052
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0008_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0009_1_SIM.cmd.err.0 <==
+ echo 1675863552
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0009_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0010_1_SIM.cmd.err.0 <==
+ echo 1675862217
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0010_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0011_1_SIM.cmd.err.0 <==
+ echo 1675863386
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0011_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0012_1_SIM.cmd.err.0 <==
+ echo 1675863720
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0012_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0013_1_SIM.cmd.err.0 <==
+ echo 1675862719
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0013_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0014_1_SIM.cmd.err.0 <==
+ echo 1675862048
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0014_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0015_1_SIM.cmd.err.0 <==
+ echo 1675865055
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0015_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0016_1_SIM.cmd.err.0 <==
+ echo 1675864721
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0016_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0017_1_SIM.cmd.err.0 <==
+ echo 1675862552
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0017_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0018_1_SIM.cmd.err.0 <==
+ echo 1675863886
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0018_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0019_1_SIM.cmd.err.0 <==
+ echo 1675860898
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0019_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0020_1_SIM.cmd.err.0 <==
+ echo 1675863219
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0020_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0021_1_SIM.cmd.err.0 <==
+ echo 1675864054
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0021_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0022_1_SIM.cmd.err.0 <==
+ echo 1675862385
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0022_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0023_1_SIM.cmd.err.0 <==
+ echo 1675864221
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0023_1_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0024_1_SIM.cmd.err.0 <==
+ echo 1675864555
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0024_1_SIM_COMPLETED
exit 0
+ exit 0

```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 8, 2023, 15:27

nvm, the second level has the same issue


```
==> a458_19600101_fc0001_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0002_2_SIM.cmd.err.0 <==
+ echo 1675865247
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0002_2_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0003_2_SIM.cmd.err.0 <==
+ echo 1675865490
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0003_2_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0004_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0005_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0006_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0007_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0008_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0009_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0010_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0011_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0012_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0013_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0014_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0015_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0016_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0017_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0018_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0019_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0020_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0021_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0022_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0023_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0024_2_SIM.cmd.err.0 <==
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
bsc32070@login2:/gpfs/scratch/bsc32/bsc32340/a458/LOG_a458> 
```

Could you do a sacct ?

It is weird because all horizontal level finished, 

```
a458_19600101_fc0024_1_SIM.cmd.out.0

2023-02-08 14:28:05.789585 The job  a458_19600101_fc0000_1_SIM.cmd  has been COMPLETED
2023-02-08 14:28:05.789892 The job  a458_19600101_fc0001_1_SIM.cmd  has been COMPLETED
2023-02-08 14:28:05.789990 The job  a458_19600101_fc0002_1_SIM.cmd  has been COMPLETED
2023-02-08 14:28:05.790077 The job  a458_19600101_fc0003_1_SIM.cmd  has been COMPLETED
2023-02-08 14:28:05.790161 The job  a458_19600101_fc0004_1_SIM.cmd  has been COMPLETED
2023-02-08 14:53:08.390291 The job  a458_19600101_fc0005_1_SIM.cmd  has been COMPLETED
2023-02-08 15:01:29.894737 The job  a458_19600101_fc0006_1_SIM.cmd  has been COMPLETED
2023-02-08 15:01:29.896492 The job  a458_19600101_fc0007_1_SIM.cmd  has been COMPLETED
2023-02-08 15:01:29.896904 The job  a458_19600101_fc0008_1_SIM.cmd  has been COMPLETED
2023-02-08 15:01:29.897290 The job  a458_19600101_fc0009_1_SIM.cmd  has been COMPLETED
2023-02-08 15:01:29.897652 The job  a458_19600101_fc0010_1_SIM.cmd  has been COMPLETED
2023-02-08 15:01:29.897981 The job  a458_19600101_fc0011_1_SIM.cmd  has been COMPLETED
2023-02-08 15:01:29.898078 The job  a458_19600101_fc0012_1_SIM.cmd  has been COMPLETED
2023-02-08 15:01:29.898481 The job  a458_19600101_fc0013_1_SIM.cmd  has been COMPLETED
2023-02-08 15:01:29.898870 The job  a458_19600101_fc0014_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.967077 The job  a458_19600101_fc0015_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.967549 The job  a458_19600101_fc0016_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.967655 The job  a458_19600101_fc0017_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.967751 The job  a458_19600101_fc0018_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.968583 The job  a458_19600101_fc0019_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.969130 The job  a458_19600101_fc0020_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.969237 The job  a458_19600101_fc0021_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.969954 The job  a458_19600101_fc0022_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.970141 The job  a458_19600101_fc0023_1_SIM.cmd  has been COMPLETED
2023-02-08 15:04:15.970238 The job  a458_19600101_fc0024_1_SIM.cmd  has been COMPLETED
```

Then the next level started:

```
a458_19600101_fc0000_2_SIM.cmd.out.0
a458_19600101_fc0001_2_SIM.cmd.out.0

a458_19600101_fc0002_2_SIM.cmd.out.0


a458_19600101_fc0003_2_SIM.cmd.out.0
a458_19600101_fc0004_2_SIM.cmd.out.0


a458_19600101_fc0005_2_SIM.cmd.out.0

a458_19600101_fc0006_2_SIM.cmd.out.0

a458_19600101_fc0007_2_SIM.cmd.out.0

a458_19600101_fc0008_2_SIM.cmd.out.0

a458_19600101_fc0009_2_SIM.cmd.out.0

a458_19600101_fc0010_2_SIM.cmd.out.0

a458_19600101_fc0011_2_SIM.cmd.out.0

a458_19600101_fc0012_2_SIM.cmd.out.0
a458_19600101_fc0013_2_SIM.cmd.out.0


a458_19600101_fc0014_2_SIM.cmd.out.0

a458_19600101_fc0015_2_SIM.cmd.out.0

a458_19600101_fc0016_2_SIM.cmd.out.0

a458_19600101_fc0017_2_SIM.cmd.out.0

a458_19600101_fc0018_2_SIM.cmd.out.0
a458_19600101_fc0019_2_SIM.cmd.out.0


a458_19600101_fc0020_2_SIM.cmd.out.0

a458_19600101_fc0021_2_SIM.cmd.out.0
a458_19600101_fc0022_2_SIM.cmd.out.0


a458_19600101_fc0024_2_SIM.cmd.out.0
a458_19600101_fc0023_2_SIM.cmd.out.0
```

However, they are getting this error.

```
==> a458_19600101_fc0000_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0001_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0002_2_SIM.cmd.err.0 <==
+ echo 1675865247
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0002_2_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0003_2_SIM.cmd.err.0 <==
+ echo 1675865490
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0003_2_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0004_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0005_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0006_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0007_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0008_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0009_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0010_2_SIM.cmd.err.0 <==
+ echo 1675865740
touch ${job_name_ptrn}_COMPLETED
+ touch /gpfs/scratch/bsc32/bsc32340/a458/LOG_a458/a458_19600101_fc0010_2_SIM_COMPLETED
exit 0
+ exit 0

==> a458_19600101_fc0011_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0012_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0013_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Step created for job 27340979

==> a458_19600101_fc0014_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0015_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0016_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0017_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0018_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0019_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0020_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0021_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0022_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0023_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)

==> a458_19600101_fc0024_2_SIM.cmd.err.0 <==
srun: Job 27340979 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27340979 step creation still disabled, retrying (Requested nodes are busy)
```

I see now that some of them have been completed, but they were getting the same issue a few moments ago.. so it looks like the template code is not properly releasing the resources for the next level


Btw, this experiment doesn't have tmp/LOG_a458 , I think it gonna mess with completed files and logs

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 8, 2023, 16:00

Yes, sorry about the tmp/LOG_, I explained above that I removed it by "accident". That's why I created a new exp a5nd with the same parameters... let's see. Now have to leave, tomorrow morning will check again. Gràcies!

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 8, 2023, 16:01

More SIM jobs are being completed, but their start is delayed. Same error. The node is busy, but eventually, they manage to start. 


Okay, I think I figured out the issue, 

The wrapper is not in --exclusive mode, so more users are using your node. When the first level finishes, the next one does not have all resources available because other HPC users are using it.

There is not currently a way to set --exclusive to a wrapper, only at job level and I think it is not enough. 

The solution is then:

* Use horizontal wrapper, up to 48 jobs ( so it uses one node ).
* Use horizontal-vertical with the same settings but increase the wallclock time of the SIM jobs, so they don't fail before they start ( 20 minutes) and use `RETRIALS = 1` so when they be re-launched when they fail by wallclock

For autosubmit 3.15, I'll add the CUSTOM_DIRECTIVE parameter to wrapper so we can use the --exclusive

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 8, 2023, 16:03

Great, thanks!

BTW, in another experiment (a5mr) I used only horizontal wrappers, and this is the one that compelted the fastest. 

What are then the optimal AS parameters for an experiment with horizontal wrappers only? thanks...

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 8, 2023, 16:06

```
[wrapper]
TYPE = horizontal
JOBS_IN_WRAPPER = SIM
MIN_WRAPPED = 2
MAX_WRAPPED = 48
```
```
[SIM]
TASKS = 1
```
```
[MARENOSTRUM4]
PROCESSORS_PER_NODE = 48
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 8, 2023, 16:11

Oh wait, we can add the --exclusive at platform level. It will be applied to all jobs but shouldn't be a problem for this experiment. To do so:

```
[marenostrum4]
# Queue type. Options: ps, SGE, LSF, SLURM, PBS, eceaccess
TYPE = slurm
HOST = mn1.bsc.es
PROJECT = bsc32
USER = bsc32340 
SCRATCH_DIR = /gpfs/scratch
ADD_PROJECT_TO_HOST = false
QUEUE = debug
PROCESSORS_PER_NODE = 48
MAX_PROCESSORS = 720
MAX_WALLCLOCK = 02:00
CUSTOM_DIRECTIVES = ["#SBATCH --exclusive"]

```

With this the horizontal-vertical should work


tested at: 

```
###############################################################################
#              a058_ASThread_16758689425478_96_2
###############################################################################
#
#SBATCH -J a058_ASThread_16758689425478_96_2
#SBATCH --qos=debug
#SBATCH -A bsc32
#SBATCH --output=a058_ASThread_16758689425478_96_2.out
#SBATCH --error=a058_ASThread_16758689425478_96_2.err
#SBATCH -t 00:10:00
#SBATCH --cpus-per-task=1
#SBATCH -n 96
#
#SBATCH --exclusive
#
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 8, 2023, 20:15

Sorry for the long silence. As said in the other issue, these are very busy times for us :sweat_smile: . You did great debugging work @dbeltrankyl.

Just to say that in the beginning, I had a hard time getting the picture... because I remember Marcus' workflow had a different than usual logic of dependencies. He was using chunks for the horizontal and members in the vertical and I wasn't sure if that's still the case, but it seems it isn't from what I see in the GUI for `a45b`.

One observation. It seems that your SIMs are only taking 3 minutes each... I think that you could increase the `MAX_WRAPPED_V`, don't you think so?

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 9, 2023, 07:05

Hi all,

Thanks @dbeltrankyl for you latest suggestions and @mcastril for joining the conversation! Highly appreciated given how busy you are :smile: . Needless to say, you will be coauthors of the next paper I prepare...

Just a clarification:

- In the workflow this issue refers too, used for systematic exploration of the parameter space for **sensitivity analysis: members are horizontal and chunks are vertical**.

- However, the templates used here are a modification of the **genetic algorithm (GA)** workflow developed by Marcus Falls. **In the original GA workflow chunks were horizontal and members were vertical** (you remembered right @mcastril !). With Marcus we used horizontal wrappers to run all 100 sims of each generation in parallel, whose results were then collected by the POST job before the next 100 sims started. **I am going to start running GA experiments again during the next few weeks**. Some debugging may be needed but I'm sure it will be very effective after all the work we did here with @dbeltrankyl .

Will continue with the testing today and tell you as soon as something requires your attention. I think we are getting there! thanks again

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 9, 2023, 09:25

Well, let's continue. Early this morning I launched

- a5nd with horizontal wrapper parameters suggested above
- a458 with horizontal-vertical wrapper parameters suggested above

Both failed...

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 9, 2023, 09:48

Hello, 

> One observation. It seems that your SIMs are only taking 3 minutes each... I think that you could increase the `MAX_WRAPPED_V`, don't you think so?

Well, by default, it is up to max_wallclock so there is no need

> * a5nd with horizontal wrapper parameters suggested above
> * a458 with horizontal-vertical wrapper parameters suggested above


Yes, it seems that is the same busy error, some jobs run, and some jobs are being canceled due to wallclock.

> * a458 with horizontal-vertical wrapper parameters suggested above

I see that the jobs were running in exclusive mode in the case of the a458. Not sure what the issue may be. I don't notice any problem with the wrapper itself. The parameters seem correct, and also the list of jobs. 

> * a5nd with horizontal wrapper parameters suggested above

And the a5nd has the same issue but doesn't make sense because it uses a horizontal wrapper, and yesterday, all first 25 members of the horizontal wrapper were completed without issues.


One thing is for sure, we can discard that due to other users using the node. 


@mcastril do you think it could be due to the job.cmd itself? maybe the mpirun -np 1 ./nemo is using more resources than the specified due some hidden flag?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 9, 2023, 09:58

Also, this issue is always raised after 

```
mpirun -np 1 ./nemo
+ mpirun -np 1 ./nemo
srun: Job 27345709 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27345709 step creation still disabled, retrying (Requested nodes are busy)
```
The rest of the template always runs fine, and other models works with these kind of wrappers

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 10, 2023, 19:38

> Thanks @dbeltrankyl for you latest suggestions and @mcastril for joining the conversation! Highly appreciated given how busy you are :smile: . Needless to say, you will be coauthors of the next paper I prepare...

Thanks @mgali, we appreciate that!

We had a separate discussion about this issue in our team meeting yesterday. The wrapper works fine with a dummy experiment (it runs sleep commands instead of real workloads). So it might be something related to the `mpi` environment as @dbeltrankyl points out.

1) In general, users running horizontal or horizontal-vertical wrappers use `machinefiles`. This is to give a hint to the MPI runtime of where each process will be located. Maybe worth to give it a go? I show the Auto-MONARCH case as an example:

https://earth.bsc.es/gitlab/es/auto-monarch/-/blob/master/templates/05b_sim.sh#L804

That line should be added after the `mpirun` and before the rest of arguments, so that `mpirun` reads the `machinefile` (automatically generated by Autosubmit).

2) Another option is to use a special type of horizontal-wrappers for intra-node allocations that use `srun` instead of `mpirun`:

https://autosubmit.readthedocs.io/en/master/userguide/wrappers/index.html#method

If none of these approaches work, @dbeltrankyl would have to copy your experiment and run it while logging in to one of the compute nodes to figure out what's going on.

We are not sure of why this set up worked for Marcus' experiments and doesn't work now. Environment changes? Could be. We have had issues running `DDT debugger` with `mpirun` lately, which was working perfectly in the past, and operations asked us to move to `srun`.

**Comment by kinow** [comment]
In GitLab by @mgali on Feb 11, 2023, 07:10

Hi,

Thanks for taking care of this and for your inputs. I just checked one of the latest experiments and of course found the error mentioned by @dbeltrankyl above. What I find curious is that when I look at job `a5nh_19600101_fc0002_1_SIM`, for example, I see in the runtime directory that it was running. But it was terminated with that error. See 
`/gpfs/scratch/bsc32/bsc32340/a5nh/LOG_a5nh/a5nh_19600101_fc0002_1_SIM.cmd.err.1`.

PROPOSED PLAN

I am going to run this workflow only a few times. So perhaps it is better that I run it with suboptimal wrappers or even sequentially (worst case scenario, will take just a week) to not abuse of your time.

I will get back to you in 1-2 weeks when I test the **genetic algorithm** workflow. I will have to run several genetic algorithm experiments and in this case it will be more important that the workflow runs smoothly. Sounds good?

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 15, 2023, 17:38

> I am going to run this workflow only a few times. So perhaps it is better that I run it with suboptimal wrappers or even sequentially (worst case scenario, will take just a week) to not abuse of your time.

I'd say that you can try running with:

```
[marenostrum4]
PROCESSORS_PER_NODE = 24
```

As @dbeltrankyl said. 

I don't think it's totally inefficient. NEMO is a memory-bound code and I think that using 24 cores per node, your runs should finish a little bit earlier, as they would have more memory bandwidth per core.

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 2, 2023, 10:14

marked this issue as related to #949

**Comment by kinow** [comment]
In GitLab by @mgali on Mar 17, 2023, 11:52

mentioned in issue cp/genetic_algorithm_pisces1d#11
